modules/ding_gatewayf/ding_gatewayf.module

```diff
@@ -241,6 +246,70 @@ function ding_gatewayf_callback() {
 }
 
 /**
+ * Do a check on the checksum given from gatewayf.
```

Comment:
we updated the gateway to deliver a checksum and basically duplicated the methods to get a match, so we know (or at least hope) the implementation is correct because we did it ourselves